fixing #1 docker for local dev, fixing ruby ver

BretFisher opened the pull request:
- matches ruby version to dockerfile version and gemfile version (with latest ruby image hotfix)
- adds .dockerignore to prevent unneeded files in image
- updates dockerfile to ensure all files are added and proper CMD is used
- updates docker-compose file to set proper envvar, bind mount needed dirs and files, and set custom entrypoint and cmd for development
- creates docker-entrypoint-dev.sh to solve local dev-specific issues before ruby is started
- removes need for .env file by having them set in compose file and pulled into database.yml

Let me know if you have questions!

g0ld2k commented:
@BretFisher I am trying to follow the Docker instructions from the README and am getting an error when running the docker-compose to setup the Postgres database.

```
docker-compose run --rm web bundle exec rake db:setup
Creating adoptadrain_db_1 ...
Creating adoptadrain_db_1 ... done
rake aborted!
No Rakefile found (looking for: rakefile, Rakefile, rakefile.rb, Rakefile.rb)
/usr/local/bundle/gems/rake-12.1.0/exe/rake:27:in `<top (required)>'
(See full trace by running task with --trace)
```

Were you able to start the docker containers, setup the db, seed data, and be able to access the web app?

BretFisher commented:
Hi Chris, My guess is you need to update your `web` image build, so first `docker-compose build` then try the db:setup.

Also, because you may be switching back and forth between branches and wanting to test this PR as a "new user" you should likely first `docker-compose down -v` to ensure there's nothing lingering around like db volumes that could affect you testing this.

BretFisher commented:
Added a few more tweaks like a "named volume" to ensure db's stay around between `docker-compose up` and `docker-compose down`. Use `docker-compose down -v` to remove them :)

Also changed postgres to use high port in compose file incase you also have it running on host or another container...

Lastly, added some ignores and one for `*override*` so an individual developer can create a `docker-compose.override.yml` to set custom local dev settings over top of `docker-compose.yml`. See how docker recommends [to use this layered approach for each local dev](https://docs.docker.com/compose/extends/#adding-and-overriding-configuration).

g0ld2k commented:
@BretFisher Thanks 😎 I'm just at the beginning of understanding Docker but definitely see the power of it.

BretFisher commented:
Shameless plug for https://bretfisher.com/dockermastery which teaches you the full dev-test-prod process :)
